foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/database/Oracle8Platform.java

```diff
@@ -14,12 +14,274 @@
+     * method returns a wrapper object which knows whether it should be bound or
+     * appended and knows how to do that.
+     */
+    @Override
+    public Object getCustomModifyValueForCall(Call call, Object value, DatabaseField field, boolean shouldBind) {
+        Class type = field.getType();
+        if (ClassConstants.BLOB.equals(type) || ClassConstants.CLOB.equals(type)) {
+            if (value == null) {
+                return null;
+            }
+            value = convertToDatabaseType(value);
+            if (shouldUseLocatorForLOBWrite()) {
+                if (lobValueExceedsLimit(value)) {
+                    ((DatabaseCall) call).addContext(field, value);
+                    if (ClassConstants.BLOB.equals(type)) {
+                        if (shouldBind) {
+                            value = new byte[1];
+                        } else {
+                            value = new SimpleAppendCallCustomParameter("empty_blob()");
+                        }
+                    } else {
+                        if (shouldBind) {
+                            value = new String(" ");
+                        } else {
+                            value = new SimpleAppendCallCustomParameter("empty_clob()");
+                        }
+                    }
+                }
+            }
+            return value;
+        }
+        return super.getCustomModifyValueForCall(call, value, field, shouldBind);
+    }
+
+    
+    /**
+     * INTERNAL: Write LOB value - only on Oracle8 and up
+     */
+    @SuppressWarnings("deprecation")
+    @Override
+    public void writeLOB(DatabaseField field, Object value, ResultSet resultSet, AbstractSession session)
+            throws SQLException {
+        if (isBlob(field.getType())) {
+            // change for 338585 to use getName instead of getNameDelimited
+            java.sql.Blob blob = (java.sql.Blob) resultSet.getObject(field.getName());
+            blob.setBytes(1, (byte[]) value);
+            //impose the localization
+            session.log(SessionLog.FINEST, SessionLog.SQL, "write_BLOB", Long.valueOf(blob.length()), field.getName());
+        } else if (isClob(field.getType())) {
+            // change for 338585 to use getName instead of getNameDelimited
+            java.sql.Clob clob = (java.sql.Clob) resultSet.getObject(field.getName());
+            clob.setString(1, (String) value);
+            //impose the localization
+            session.log(SessionLog.FINEST, SessionLog.SQL, "write_CLOB", Long.valueOf(clob.length()), field.getName());
+        } else {
+            // do nothing for now, open to BFILE or NCLOB types
+        }
+    }
+
+    /**
+     * INTERNAL: Used in writeLOB method only to identify a BLOB
+     */
+    protected boolean isBlob(Class type) {
+        return ClassConstants.BLOB.equals(type);
+    }
+
+    /**
+     * INTERNAL: Used in writeLOB method only to identify a CLOB
+     */
+    protected boolean isClob(Class type) {
```

Comment:
hmm, `isClob` is overridden by  org.eclipse.persistence.platform.database.oracle.Oracle9Platform. However, that change includes this impl ( ... || super.isClob(type)), so perhaps it isnt 100% needed. I was going to suggest porting Oracle9Platform.isClob() impl as well, but that seems like a larger can of worms (N-Type support?)... Do we need NClob support as well for this LOB Location support?

Reply:
That has a dependency on org.eclipse.persistence.platform.database.oracle.NClob -- not sure we want to port that at this time.